src/flows/components/header/Submit.tsx

```diff
@@ -16,130 +9,51 @@ import {
   List,
   ButtonGroup,
 } from '@influxdata/clockface'
-import {useDispatch} from 'react-redux'
 
 import {SubmitQueryButton} from 'src/timeMachine/components/SubmitQueryButton'
 import {QueryContext} from 'src/flows/context/query'
-import {FlowContext} from 'src/flows/context/flow.current'
 import {RunModeContext, RunMode} from 'src/flows/context/runMode'
-import {ResultsContext} from 'src/flows/context/results'
 import {notify} from 'src/shared/actions/notifications'
-import {PIPE_DEFINITIONS} from 'src/flows'
+import {FlowContext} from 'src/flows/context/flow.current'
 
 // Utils
-import {event} from 'src/cloud/utils/reporting'
 import {cancelAllRunningQueries} from 'src/timeMachine/actions/queries'
-
-// Constants
-import {
-  notebookRunSuccess,
-  notebookRunFail,
-} from 'src/shared/copy/notifications'
-import {PROJECT_NAME} from 'src/flows'
-
-// Types
-import {RemoteDataState} from 'src/types'
+import {PIPE_DEFINITIONS} from 'src/flows'
 
 // Styles
 import 'src/flows/components/header/Submit.scss'
 
 const fakeNotify = notify
 
 export const Submit: FC = () => {
-  const dispatch = useDispatch()
-  const {query, generateMap} = useContext(QueryContext)
-  const {runMode, setRunMode} = useContext(RunModeContext)
   const {flow} = useContext(FlowContext)
-  const {add, update} = useContext(ResultsContext)
-  const [isLoading, setLoading] = useState(RemoteDataState.NotStarted)
+  const {runMode, setRunMode} = useContext(RunModeContext)
+  const {isLoading, queryAll} = useContext(QueryContext)
+
+  const _range = useMemo(
+    () => `${flow.range.lower} to ${flow.range.upper || 'now'}`,
+    [flow.range]
+  )
 
   const hasQueries = useMemo(() => {
     return !!flow.data.all
+      .filter(p => {
+        // this condition ensures that empty metric selectors are filtered out
+        if (p?.type === 'metricSelector') {
+          return !!p?.bucket
+        }
+        return true
+      })
```

Comment:
nit: could be a ternary if you want. `return p?.type === 'metricSelector' ? !!p?.bucket : true`